servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/TimeoutFromRequest.java

```diff
@@ -37,10 +42,24 @@
      *
      * @param request the current request
      * @return The timeout or null for no timeout
+     * @deprecated Use {@link #apply(HttpRequestMetaData, TimeSource)}. This method will be removed in a future release.
      */
     @Override
+    @Deprecated
+    @Nullable
+    default Duration apply(HttpRequestMetaData request) {
```

Comment:
There was a discussion to remove this interface because we agreed that all user-provided funtrions have to be non-blocking => we don't need `ExecutionStrategyInfluencer` here => we can use `java.util.Function`.
//cc @bondolo 

We can consider doing it at the same time you need to change it for gRPC in this PR or in a follow-up to use `BiFunction`.

Reply:
We can't bring the `TimeSource` change to 0.41, so anything that has to do with `BiFunction` in this context would not make sense in 0.41. Therefore, the plan would look like the following:

1. Deprecate `TimeoutFromRequest` and usages and introduce variants accepting `Function` instead. Merge to main and cherry-pick to 0.41.
2. Pick up that change in this branch and remove `TimeoutFromRequest`, introduce variants for `BiFunction` and deprecate all `Function` variants.
3. In 0.43 we can remove `Function` variants.

Would that accomplish what you had in mind?

Reply:
The proposed plan sounds good to me.

Reply:
During the offline discussion we agreed to deprecate `TimeoutFromRequest` in 0.42 only, deprecate ctros that take `TimeoutFromRequest`. Instead, we will add ctors that take `BiFunction<HttpRequestMetaData, TimeSource, Duration>`. The `TimeoutFromRequest` will be removed in 0.43.

Reply:
And now that's been implemented.